packages/integration-sdk-core/src/types/spec.ts

```diff
@@ -1,18 +1,24 @@
 import { IntegrationInvocationConfig } from './config';
-import { IntegrationStepExecutionContext } from './context';
+import {
+  IntegrationExecutionConfig,
+  IntegrationStepExecutionContext,
+} from './context';
 import { IntegrationInstanceConfig } from './instance';
 import { Step } from './step';
 
-export interface StepSpec<TConfig extends IntegrationInstanceConfig>
-  extends Omit<
-    Step<IntegrationStepExecutionContext<TConfig>>,
+export interface StepSpec<
+  TInstanceConfig extends IntegrationInstanceConfig,
+  TExecutionConfig extends IntegrationExecutionConfig,
```

Comment:
This is a breaking change unless we fall back to `IntegrationExecutionConfig` as the default. The AWS integration is using `StepSpec` heavily e.g. https://github.com/JupiterOne/graph-aws/blob/d402343227e24395b57725924dc60ac5a403fe5c/docs/spec/src/accessanalyzer/index.ts#L4